### PR TITLE
Update images on 'Fetch data dynamically' page

### LINKED ISSUE #3254: Update images on 'Fetch data dynamically' page
Page URL: https://dart.dev/tutorials/web/fetch-data
Page source: https://github.com/dart-lang/site-www/tree/master/src/_tutorials/web/fetch-data.md

The images on the page use the old `dartlang.org` link, `http` instead of `https`, can be blurry on some screen sizes, and could use a style to match better with the rest of the website. We could create SVGs using a more user accessible tool such as https://www.diagrams.net/

### PR DESCRIPTION
**Pull request by Mayank-9018**
Fixes #3254

### CODE REVIEW

File: src/_tutorials/web/images/client-server.svg [hunk 1]
```diff
@@ -0,0 +1,3 @@
...
...xMi44YzAsMS42LTEuNiwyLjQtNCwyLjQiLz4mI3hhOzxwb2x5Z29uIHN0eWxlPSJmaWxsOiNEOEUzRUE7IiBwb2ludHM9IjI1MiwyNDAuOCAyNTYuOCwyNDUuNiAyNjAsMjQyLjQgMjUyLDIzNC40IDI0NCwyNDIuNCAyNDcuMiwyNDUuNiAiLz4mI3hhOzxjaXJjbGUgc3R5bGU9ImZpbGw6IzI3RDY4QTsiIGN4PSIyNTIiIGN5PSIxMy42IiByPSIzLjIiLz4mI3hhOzxwYXRoIHN0eWxlPSJmaWxsOiMxNUFGODM7IiBkPSJNMjUyLDEwLjRjMS42LDAsMy4yLDEuNiwzLjIsMy4ycy0xLjYsMy4yLTMuMiwzLjIiLz4mI3hhOzxlbGxpcHNlIHN0eWxlPSJmaWxsOiNDRkZGRUY7IiBjeD0iMjUwLjM3MyIgY3k9IjEyLjc4MiIgcng9IjAuOCIgcnk9IjAuOCIvPiYjeGE7PGc+JiN4YTs8L2c+JiN4YTs8Zz4mI3hhOzwvZz4mI3hhOzxnPiYjeGE7PC9nPiYjeGE7PGc+JiN4YTs8L2c+JiN4YTs8Zz4mI3hhOzwvZz4mI3hhOzxnPiYjeGE7PC9nPiYjeGE7PGc+JiN4YTs8L2c+JiN4YTs8Zz4mI3hhOzwvZz4mI3hhOzxnPiYjeGE7PC9nPiYjeGE7PGc+JiN4YTs8L2c+JiN4YTs8Zz4mI3hhOzwvZz4mI3hhOzxnPiYjeGE7PC9nPiYjeGE7PGc+JiN4YTs8L2c+JiN4YTs8Zz4mI3hhOzwvZz4mI3hhOzxnPiYjeGE7PC9nPiYjeGE7PC9zdmc+" preserveAspectRatio="none"/></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (parlough) [inline src/_tutorials/web/images/client-server.svg:3]: If we could make this one somehow take up a bit less vertical space that may be good.

File: src/_tutorials/web/images/portmanteaux-code.svg [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
...troke-miterlimit="10" pointer-events="none"/><path d="M 212.2 559.16 L 212 394.12" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 212 391.12 L 214.01 395.12 L 212 394.12 L 210.01 395.12 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 213.53 600.6 L 247 600.14 Q 257 600 257 590 L 257 394.96" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 257 391.96 L 259 395.96 L 257 394.96 L 255 395.96 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 412 629.66 L 437 629.9 Q 447 630 447 620 L 447 415.62" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 447 411.12 L 450 417.12 L 447 415.62 L 444 417.12 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (parlough) [inline src/_tutorials/web/images/portmanteaux-code.svg:3]: Could you update the code to match the updated tutorial?

```dart
Future<void> makeRequest(Event _) async {
  const path = 'https://dart.dev/f/portmanteaux.json';
  final httpRequest = HttpRequest();
  httpRequest
    ..open('GET', path)
    ..onLoadEnd.listen((e) => requestComplete(httpRequest))
    ..send('');
}
```

Also:

1. When I'm viewing it, it seems the code block is hanging outside the background. Feel free to make the background larger to get it to fit.
2. As for the Portmanteaux application example, I think it's fine in this case that the application image is not an actual image for size purposes. But could you update the button text to say "Get portmanteaux" and perhaps use a rounded rectangle for the button to match the current UI(I believe Diagrams.net has this as a built in shape, but if not the rectangle is fine).

DartPad rendered version for reference:
image

File: src/_tutorials/web/images/set-callback.svg [hunk 3]
```diff
@@ -0,0 +1,3 @@
...
...><switch><foreignObject style="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 188px; height: 1px; padding-top: 115px; margin-left: 111px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 12px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><div>set callback function</div><div>(to call when request completes)</div></div></div></div></foreignObject><text x="205" y="119" fill="#000000" font-family="Helvetica" font-size="12px" text-anchor="middle">set callback function...</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (parlough) [inline src/_tutorials/web/images/set-callback.svg:3]: Could we expand the box a bit to the right? It seems the end has some strange artifacts:

image

Also a nit: There seems to be an extra space before `requestComplete`. It's okay if you can't remove it:

image

File: src/_tutorials/web/images/uri-details.svg [hunk 4]
```diff
@@ -0,0 +1,3 @@
...
...none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe center; width: 1px; height: 1px; padding-top: 135px; margin-left: 80px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 14px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; font-weight: bold; white-space: nowrap; "><span style="font-size: 14px ; font-weight: normal"><font style="font-size: 14px ; background-color: rgb(164 , 253 , 223)">path to resource</font></span></div></div></div></foreignObject><text x="80" y="139" fill="#000000" font-family="Helvetica" font-size="14px" text-anchor="middle" font-weight="bold">path to resource</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (parlough) [inline src/_tutorials/web/images/uri-details.svg:3]: This is great, thanks!

**Review by parlough — COMMENTED** [review]
Thanks so much for working on this, these are great. I left a few comments, but if anything seems like not a good suggestion or hard to implement with the tool, just let me know and that's completely fine!

Since we may optimize SVGs in the `src` directories in the future(which would remove the diagrams.net metadata), we are storing the original copies in the top level `/diagrams/` directory. Could you create a new directory within `/diagrams/tutorials/web/` and store original copies there?

**Comment by Mayank-9018** [comment]
A lot of issues like overlapping and weird shapes may have been caused by different break types, git converted `lf` to `crlf`. I'll see to it.

File: src/_tutorials/web/images/portmanteaux-code.svg [hunk 5]
```diff
@@ -0,0 +1,3 @@
...
...: flex; align-items: unsafe center; justify-content: unsafe center; width: 138px; height: 1px; padding-top: 130px; margin-left: 351px;"><div style="box-sizing: border-box; font-size: 0; text-align: center; "><div style="display: inline-block; font-size: 14px; font-family: Helvetica; color: #333333; line-height: 1.2; pointer-events: none; white-space: normal; word-wrap: normal; "><font style="font-size: 15px">mouse-click handler</font></div></div></div></foreignObject><text x="420" y="134" fill="#333333" font-family="Helvetica" font-size="14px" text-anchor="middle">mouse-click handler</text></switch></g><path d="M 343.66 129.43 L 73.35 105" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 348.89 129.9 L 341.6 132.76 L 343.66 129.43 L 342.23 125.78 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
Comment (parlough) [inline src/_tutorials/web/images/portmanteaux-code.svg:3]: Could you perhaps make the code box a little larger? Seems like the beginning and end are cutting off causing some artifacts.

image

**Review by parlough — APPROVED** [review]
Looks pretty good to me, thanks again for working on this!

One last suggestion but still an improvement without addressing it!

**Comment by Mayank-9018** [comment]
Issues like overlapping and text running out of box were caused by the fonts. To fix them, more web-safe fonts like _Helvetica_ and _Courier new_ are now used. @parlough Can you confirm the images look exactly like these https://imgur.com/a/bXR720N ?

**Comment by parlough** [comment]
@Mayank-9018 Thanks for the continued work and deep consideration around that. The images match and the arrows are appearing as expected! :partying_face:

**Review by kwalrath — COMMENTED** [review]
Sorry for the delay in reviewing this! It's the kind of detailed, side-by-side comparison that takes concentration to do justice to... and I finally got that chance to concentrate.

I have a bunch of nitpicky comments, but in general I love the new figures. Thank you for creating them, @mayank-9018! I think we're very close to having versions we can publish. OK, here come the excruciatingly detailed comments.

File: diagrams/tutorials/web/jsonstring.svg [hunk 6]
```diff
@@ -0,0 +1,3 @@
...
...le="overflow: visible; text-align: left;" pointer-events="none" width="100%" height="100%" requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"><div xmlns="http://www.w3.org/1999/xhtml" style="display: flex; align-items: unsafe center; justify-content: unsafe flex-start; width: 88px; height: 1px; padding-top: 150px; margin-left: 430px;"><div style="box-sizing: border-box; font-size: 0; text-align: left; "><div style="display: inline-block; font-size: 14px; font-family: Helvetica; color: #000000; line-height: 1.2; pointer-events: all; white-space: normal; word-wrap: normal; "><span style="background-color: rgb(255 , 214 , 165) ; font-size: 14px"><font style="font-size: 14px">List of strings</font></span></div></div></div></foreignObject><text x="430" y="154" fill="#000000" font-family="Helvetica" font-size="14px">List of strin...</text></switch></g></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
[the remark this answers is not in the document]
Comment (Mayank-9018) [reply diagrams/tutorials/web/jsonstring.svg:3]: Something like this?

The grey shade(used in Map:string....) was illegible here, so instead used a more darker shade.

File: src/_tutorials/web/images/client-server.svg [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
```
[the remark this answers is not in the document]
Comment (Mayank-9018) [reply src/_tutorials/web/images/client-server.svg:3]: I nearly forgot about it! Yes, the images are free but require attribution. But if that's an issue, somewhat similar images can be recreated easily without much effort. 

> To make clearer that the arrows are associated with the request & response, the boxes should be on or centered over their respective arrow lines.

I have thought of this before. Is this OK?:

File: diagrams/tutorials/web/portmanteaux-code.svg [hunk 8]
```diff
@@ -0,0 +1,3 @@
...
...2M6d29yaz4mI3hhOyAgICAgIDxjYzpsaWNlbnNlIHJkZjphYm91dD0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvcHVibGljZG9tYWluL3plcm8vMS4wLyI+JiN4YTsgICAgICAgIDxjYzpwZXJtaXRzIHJkZjpyZXNvdXJjZT0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjUmVwcm9kdWN0aW9uIi8+JiN4YTsgICAgICAgIDxjYzpwZXJtaXRzIHJkZjpyZXNvdXJjZT0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjRGlzdHJpYnV0aW9uIi8+JiN4YTsgICAgICAgIDxjYzpwZXJtaXRzIHJkZjpyZXNvdXJjZT0iaHR0cDovL2NyZWF0aXZlY29tbW9ucy5vcmcvbnMjRGVyaXZhdGl2ZVdvcmtzIi8+JiN4YTsgICAgICA8L2NjOmxpY2Vuc2U+JiN4YTsgICAgPC9yZGY6cmRmPiYjeGE7ICA8L21ldGFkYXRhPiYjeGE7PC9zdmc+" preserveAspectRatio="none" pointer-events="none"/><path d="M 353.71 149.9 L 73.35 105" fill="none" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/><path d="M 358.9 150.73 L 351.43 153.08 L 353.71 149.9 L 352.54 146.17 Z" fill="#000000" stroke="#000000" stroke-miterlimit="10" pointer-events="none"/></g><switch><g requiredFeatures="http://www.w3.org/TR/SVG11/feature#Extensibility"/><a transform="translate(0,-5)" xlink:href="https://www.diagrams.net/doc/faq/svg-export-text-problems" target="_blank"><text text-anchor="middle" font-size="10px" x="50%" y="100%">Viewer does not support full SVG 1.1</text></a></switch></svg>
```
[the remark this answers is not in the document]
Comment (Mayank-9018) [reply diagrams/tutorials/web/portmanteaux-code.svg:3]: > I wonder if we could use a cursor with lines around it.

I tried it with lines, but it didn't look as good.So, instead, I used a blue circle.
Indicating click with blue circle

File: src/_tutorials/web/images/client-server.svg (hunk 7 above)
[the remark this answers is not in the document]
Comment (Mayank-9018) [reply src/_tutorials/web/images/client-server.svg:3]: A quick one that I made; requires no attribution :)

File: diagrams/tutorials/web/portmanteaux-code.svg (hunk 8 above)
[the remark this answers is not in the document]
Comment (kwalrath) [reply diagrams/tutorials/web/portmanteaux-code.svg:3]: I like that!

File: src/_tutorials/web/images/client-server.svg (hunk 7 above)
[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_tutorials/web/images/client-server.svg:3]: Looks good!

File: diagrams/tutorials/web/jsonstring.svg (hunk 6 above)
[the remark this answers is not in the document]
Comment (kwalrath) [reply diagrams/tutorials/web/jsonstring.svg:3]: Yes, that works.

File: src/_tutorials/web/images/client-server.svg (hunk 7 above)
[the remark this answers is not in the document]
Comment (Mayank-9018) [reply src/_tutorials/web/images/client-server.svg:3]: So, which images should I use? If the second one, what more changes should be made?

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_tutorials/web/images/client-server.svg:3]: I like the second one, yours! The only changes I might make (and they aren't essential) would be:

1. Don't  use code font for the URL (per https://developers.google.com/style/code-in-text#items-to-put-in-ordinary-non-code-font).
2. In the HTTP response, make the "..." the same color as the tags, and/or don't indent as much. With the current color and indentation, the "..." is distracting.

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_tutorials/web/images/client-server.svg:3]: I just checked the figures again, and I think that once we update this one, we're good to go!

[the remark this answers is not in the document]
Comment (Mayank-9018) [reply src/_tutorials/web/images/client-server.svg:3]: I fixed the "..." in the HTTP response and changed the font of the URL, but I still prefer it to be code-fonted (it looks more uniform and clean). Your call
Changed fonts 
Original fonts

[the remark this answers is not in the document]
Comment (kwalrath) [reply src/_tutorials/web/images/client-server.svg:3]: I think you're right that the code fonted URL looks more consistent. So let's use the bottom figure.

**Review by kwalrath — APPROVED** [review]
These figures look fantastic. Thank you, @Mayank-9018, for making these and being so responsive to our reviews!

**Pull request merged**